scripts/prepare_disk.sh

```diff
@@ -22,6 +22,58 @@ usage()
     echo "NOTE: This script is destructive and will destroy your data."
 }
 
+is_integer()
+{
+    test "${1}" -eq "${1}" 2> /dev/null
+}
+
+is_comment()
+{
+    test -z "${1%%#*}"
+}
+
+# A bit counter intuitive, a function can either return an exit code,
+# or set a global variable, or echo something, in this case the
```

Comment:
This sounds a bit odd, "it can set an exit code or a global". How is the expected use? And what global does it set, and for what reason? Or is that a side-effect that should be ignored?

Reply:
Yess that is correct, but maybe i should explain it a bit better.

Reply:
was the while that was executed in the subshell or was it the earlier version where you had
```sh
sfdisk | while ...
```
which causes the sfdisk to be called in an sfdisk.

As for explaining it better, be careful not to overexplain it, some of the things are knowing how to write shell, be careful not to explain shell scripting basics (even though it is VERY tempting).